sktime/tests/tests/test_recursive_reduction.py

```diff
@@ -0,0 +1,15 @@
+import pytest
+import pandas as pd
+from sktime.forecasting.compose import RecursiveReductionForecaster
+from sktime.forecasting.naive import NaiveForecaster
+
+@pytest.fixture
+def sample_series():
+    """Sample time series data with missing values."""
+    return pd.Series([1.0, None, 3.0, None, 5.0])
+
+def test_default_imputation(sample_series):
+    base_forecaster = NaiveForecaster(strategy="mean")
+    forecaster = RecursiveReductionForecaster(base_forecaster)
```

Comment:
Unless I'm much mistaken, the first argument is expected to be a sklearn regressor. I'm not sure how will a sktime forecaster be used there, and will it help. Also naive forecaster do not use X, so I am not sure I'm clear what you are trying to do here. Can you kindly elaborate?

Reply:
The NaiveForecaster is not a regressor and does not accept lagged features or X. Instead, it predicts based on predefined naive strategies like "mean" or "last observed value," making it unsuitable as a base model in RecursiveReductionForecaster.
The base model for RecursiveReductionForecaster is typically a regressor that can predict numerical values.It leverages the regressor to model the relationship between lagged features (X) and the target variable (y).

Reply:
@PTUsumit this is not a resolved thread. That's exactly my question to you, why are you passing an instance of `NaiveForecaster` (which you created in line 12 above) to `RecursiveReductionForecaster` (in line 13) in your test case instead of passing a regressor?